runtime/near-vm-runner/src/errors.rs

```diff
@@ -33,6 +67,84 @@ pub enum PrepareError {
     Memory,
 }
 
+impl From<wasmer_runtime::error::Error> for VMError {
+    fn from(err: wasmer_runtime::error::Error) -> Self {
+        use wasmer_runtime::error::Error;
+        match err {
+            Error::CompileError(err) => err.into(),
+            Error::LinkError(err) => VMError::FunctionCallError(FunctionCallError::LinkError(
+                format!("{}", Error::LinkError(err)),
+            )),
+            Error::RuntimeError(err) => err.into(),
+            Error::ResolveError(err) => err.into(),
+            Error::CallError(err) => err.into(),
+            Error::CreationError(err) => panic!(err),
+        }
+    }
+}
+
+impl From<CallError> for VMError {
+    fn from(err: CallError) -> Self {
+        match err {
+            CallError::Resolve(err) => err.into(),
+            CallError::Runtime(err) => err.into(),
+        }
+    }
+}
+
+impl From<CompileError> for VMError {
+    fn from(err: CompileError) -> Self {
+        VMError::FunctionCallError(FunctionCallError::CompilationError(
+            CompilationError::WasmerCompileError(err.to_string()),
+        ))
+    }
+}
+
+impl From<WasmerResolveError> for VMError {
+    fn from(err: WasmerResolveError) -> Self {
+        match err {
+            WasmerResolveError::Signature { .. } => VMError::FunctionCallError(
+                FunctionCallError::ResolveError(MethodResolveError::MethodInvalidSignature),
+            ),
+            WasmerResolveError::ExportNotFound { .. } => VMError::FunctionCallError(
+                FunctionCallError::ResolveError(MethodResolveError::MethodNotFound),
+            ),
+            WasmerResolveError::ExportWrongType { .. } => VMError::FunctionCallError(
+                FunctionCallError::ResolveError(MethodResolveError::MethodNotFound),
+            ),
+        }
+    }
+}
+
+impl From<RuntimeError> for VMError {
+    fn from(err: WasmerRuntimeError) -> Self {
+        match err {
+            WasmerRuntimeError::Trap { msg } => {
+                VMError::FunctionCallError(FunctionCallError::WasmTrap(msg.to_string()))
```

Comment:
It does not look like this code does anything specific that could address https://github.com/nearprotocol/nearcore/issues/1319 . Will Wasmer still crash the process if there is a panic inside Rust code?

Reply:
added some tests for different ways a contract can crash